### PR TITLE
llvm: update to 15.0.0

### PR DESCRIPTION
**Pull request by MehdiChinoune**
without mlir and flang.

### CODE REVIEW

File: mingw-w64-libc++/PKGBUILD [hunk 1]
```diff
@@ -154,8 +134,7 @@ package_libc++() {
              "${MINGW_PACKAGE_PREFIX}-libunwind" \
              || echo "${MINGW_PACKAGE_PREFIX}-gcc-libs"))
 
-  DESTDIR="${pkgdir}" cmake --build "${srcdir}/build-shared-${MSYSTEM}" --target install-cxx
-  DESTDIR="${pkgdir}" cmake --build "${srcdir}/build-static-${MSYSTEM}" --target install-cxx
+  DESTDIR="${pkgdir}" cmake --build "${srcdir}/build-${MSYSTEM}" --target install-cxx
 }
 
 package_libc++abi() {
```
Comment (mati865) [inline mingw-w64-libc++/PKGBUILD:140]: Let's get rid of this useless package

File: mingw-w64-clang/PKGBUILD [hunk 2]
```diff
@@ -41,70 +44,65 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
...
             '715cb8862753854b2d9256e0b70003e2d1f57083d83eaeaf5a095fc72b8a4e26'
             '3a58bd5243fb082df64c90bd9a4f00c698317f20365d11bbe30fac5dfccd215e'
-            'e6c3b97f733920b0a32e889cd180dd86507bddea09e9c217cd1445c047cd4976'
-            'bd40b6c84d25de5ac5877f2c83ac61d872838bc034df62e98ad16116d23744e0'
-            '778e0db0a5b0657ab05e2a81d83241347a4a41af2b0f9903422f651fa58e8d40'
-            'a10e5192889501c75db6a41e29df44b2af8502b3ef34c2315789563a5c8e9b85')
+            '778e0db0a5b0657ab05e2a81d83241347a4a41af2b0f9903422f651fa58e8d40')
 validpgpkeys=('B6C8F98282B944E3B0D5C2530FC3042E345AD05D'  # Hans Wennborg, Google.
-              '474E22316ABF4785A88C6E8EA2C794A986419D8A') # Tom Stellard
+              '474E22316ABF4785A88C6E8EA2C794A986419D8A'  # Tom Stellard
```
Comment (mati865) [inline mingw-w64-clang/PKGBUILD:104]: Let's remove outdated keys

File: mingw-w64-libc++/PKGBUILD (hunk 1 above)
Comment (MehdiChinoune) [reply mingw-w64-libc++/PKGBUILD:140]: Its header files are now required to build `polly`.
```
[file-diff] mingw-w64-clang-x86_64-libc++abi
  
  [MSYS2 CI] mingw-w64-libc++: File listing diff for mingw-w64-clang-x86_64-libc++abi
  
  warning: database file for 'ci' does not exist (use '-Fy' to download)
  --- /dev/fd/63	2022-09-06 19:28:31.000000000 +0000
  +++ /dev/fd/62	2022-09-06 19:28:31.000000000 +0000
  @@ -1,3 +1,8 @@
   /clang64/
  +/clang64/include/
  +/clang64/include/c++/
  +/clang64/include/c++/v1/
  +/clang64/include/c++/v1/__cxxabi_config.h
  +/clang64/include/c++/v1/cxxabi.h
   /clang64/lib/
   /clang64/lib/libc++abi.a
```

Arch Linux still have it https://archlinux.org/packages/community/x86_64/libc++abi/
see: https://pkgs.org/search/?q=libc%2B%2Babi and https://pkgs.org/download/libcxxabi

File: mingw-w64-clang/PKGBUILD (hunk 2 above)
Comment (MehdiChinoune) [reply mingw-w64-clang/PKGBUILD:104]: which ones?

File: mingw-w64-libc++/PKGBUILD (hunk 1 above)
Comment (mati865) [reply mingw-w64-libc++/PKGBUILD:140]: Oh, that's unfortunate.
On Linux `libc++.a` is a linker script that links both libc++abi and libc++ so they need to provide it. For Windows we cannot use linker scripts so we use build option that merges libc++abi and libc++ libraries into single one: `libc++.a`, so `libc++abi.a` never had any purpose for us.
You are correct about the header, I haven't noticed it was added in LLVM 15 and is the only reason for this package to exist.

File: mingw-w64-clang/PKGBUILD (hunk 2 above)
Comment (mati865) [reply mingw-w64-clang/PKGBUILD:104]: `B6C8F98282B944E3B0D5C2530FC3042E345AD05D` and `474E22316ABF4785A88C6E8EA2C794A986419D8A`

Comment (MehdiChinoune) [reply mingw-w64-clang/PKGBUILD:104]: I know Hans Wennborg doesn't involve in release process but Tom Stellard still a release manager https://discourse.llvm.org/t/llvm15-release-update/64302

**Comment by MehdiChinoune** [comment]
Is it OK to merge this?

**Comment by lazka** [comment]
(whatever the decision, don't forget the rebuilds :) )

**Comment by MehdiChinoune** [comment]
I will merge it after completing Qt5 and Qt6 updates

**Pull request merged**